src/view.rs

```diff
@@ -859,3 +859,23 @@ mod view_response_builder_tests {
         assert_eq!(expected, got);
     }
 }
+
+pub struct AllDocumentsViewValue {
```

Comment:
Ah, right, I totally misunderstood this type.

First off, forgive me because I haven't run `_all_docs` in a long time, so I'm working off memory and CouchDB documentation until I have time to dive deeper into this.

That said, shouldn't the result of `_all_docs` be something like `HashMap<DocumentId, Document>`? If so, this is another reason why `ExecuteView` may be too general for `_all_docs` because `ExecuteView` returns a `ViewResponse`, which is a run-time variant reduced/unreduced wrapper. As far as the HTTP goes, `_all_docs` is almost the same as executing a view that emits `(doc_id, doc)` for each document, but from Chill's perspective, where everything is strongly typed, `_all_docs` is more specific. I'm leaning more towards having two distinct action types—unless the code sharing between the two actions turns into a mess.

Anyway, I think you can nix `AllDocumentsViewValue` altogether.